src/debot/interfaces/encryption_box_input.rs

```diff
@@ -0,0 +1,190 @@
+	]
+}
+"#;
+
+pub struct EncryptionBoxInput {
+    handles: RwLock<Vec<TerminalEncryptionBox>>,
+    client: TonClient,
+}
+
+impl EncryptionBoxInput {
+    pub fn new(client: TonClient) -> Self {
+        Self {
+            handles: RwLock::new(vec![]),
+            client: client,
+        }
+    }
+
+    async fn get_nacl_box(&self, args: &Value) -> InterfaceResult {
+        let answer_id = decode_answer_id(args)?;
+        let prompt = decode_prompt(args)?;
+        let nonce = decode_nonce(args)?;
+        let their_pubkey = decode_string_arg(args, "theirPubkey")?;
+        println!("{}", prompt);
+        let encryption_box = TerminalEncryptionBox::new(ParamsOfTerminalEncryptionBox {
+            context: self.client.clone(),
+            box_type: EncryptionBoxType::NaCl,
+            their_pubkey: their_pubkey,
+            nonce: nonce,
+        })
+        .await;
+        let handle = encryption_box.handle();
+        self.handles.write().await.push(encryption_box);
+        Ok((answer_id, json!({ "handle": handle.0})))
+    }
+    async fn get_nacl_secret_box(&self, args: &Value) -> InterfaceResult {
+        let answer_id = decode_answer_id(args)?;
+        let prompt = decode_prompt(args)?;
+        let nonce = decode_nonce(args)?;
+        println!("{}", prompt);
+        let encryption_box = TerminalEncryptionBox::new(ParamsOfTerminalEncryptionBox {
+            context: self.client.clone(),
+            box_type: EncryptionBoxType::SecretNaCl,
+            their_pubkey: String::from(""),
+            nonce: nonce,
+        })
+        .await;
+        let handle = encryption_box.handle();
+        self.handles.write().await.push(encryption_box);
+        Ok((answer_id, json!({ "handle": handle.0})))
+    }
+    async fn get_chacha20_box(&self, args: &Value) -> InterfaceResult {
+        let answer_id = decode_answer_id(args)?;
+        let prompt = decode_prompt(args)?;
+        let nonce = decode_nonce(args)?;
+        println!("{}", prompt);
+        let encryption_box = TerminalEncryptionBox::new(ParamsOfTerminalEncryptionBox {
+            context: self.client.clone(),
+            box_type: EncryptionBoxType::ChaCha20,
+            their_pubkey: String::from(""),
+            nonce: nonce,
+        })
+        .await;
+        let handle = encryption_box.handle();
+        self.handles.write().await.push(encryption_box);
+        Ok((answer_id, json!({ "handle": handle.0})))
+    }
+    async fn remove_handle(&self, args: &Value) -> InterfaceResult {
+        let answer_id = decode_answer_id(args)?;
+        let handle = u32::from_str_radix(
+            args["handle"]
+                .as_str()
+                .ok_or(format!("handle not found in argument list"))?,
+            10,
+        )
+        .map_err(|e| format!("{}", e))
+        .unwrap();
+        self.handles
+            .write()
+            .await
+            .retain(|value| (*value).handle().0 != handle);
+        Ok((answer_id, json!({})))
```

Comment:
`remove ` method should return `removed: bool` argument

Reply:
fixed